controller/apps.go

```diff
@@ -0,0 +1,413 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AppsController implements the apps resource.
+type AppsController struct {
+	*goa.Controller
+	AuthURL string
+	WitURL  string
+}
+
+// NewAppsController creates a apps controller.
+func NewAppsController(service *goa.Service, config *configuration.Registry) *AppsController {
+	return &AppsController{
+		Controller: service.NewController("AppsController"),
+
+		AuthURL: config.GetAuthServiceURL(),
+		//AuthURL: "http://localhost:8089"
+		//AuthURL: "https://auth.prod-preview.openshift.io",
+		//AuthURL: "https://auth.openshift.io",
+
+		// TODO - make this a config variable?
+		//WitURL: "http://localhost:8080"
+		//WitURL: "http://api.prod-preview.openshift.io",
+		WitURL: "http://api.openshift.io",
+	}
+}
+
+func tostring(item interface{}) string {
+	bytes, _ := json.MarshalIndent(item, "", "  ")
+	return string(bytes)
+}
+
+func (c *AppsController) getAndCheckOsioClient(ctx context.Context) (*OsioClient, error) {
+	oc, err := NewOsioClient(ctx, c.WitURL)
+	if err != nil {
+		return nil, errors.NewUnauthorizedError("osio")
+	}
+	return oc, nil
+}
+
+func (c *AppsController) getSpaceNameFromSpaceID(ctx context.Context, spaceID uuid.UUID) (*string, error) {
+	// use WIT API to convert Space UUID to Space name
+	oc, err := c.getAndCheckOsioClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	osioSpace, err := oc.GetSpaceByID(spaceID.String(), false)
+	if err != nil {
+		return nil, errors.NewNotFoundError("osio space", spaceID.String())
+	}
+	return osioSpace.Attributes.Name, nil
+}
+
+func (c *AppsController) getNamespaceName(ctx context.Context) (*string, error) {
+	osioclient, err := c.getAndCheckOsioClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeSpaceAttr, err := osioclient.GetNamespaceByType(nil, "user")
```

Comment:
I did write code to use Tenant, but in a conversation with @aslakknutsen, he pointed out that since this service will probably be separated out from WIT, it might be better to use my WIT code for now, and then upgrade to useing Tenant later.  This (if this PR is approved) allows the other team members to start working with the new API while I address issues like this in a later PR.